tools/googlesearch/searchTool.go

```diff
@@ -0,0 +1,48 @@
+package googlesearch
+
+import (
+	"strings"
+
+	"github.com/tmc/langchaingo/tools"
+	"github.com/tmc/langchaingo/utils/serpapi"
+)
+
+type SearchTool struct {
+	name        string
```

Comment:
it's odd to export a type but then make the fields unexported, can you explain your rationale?